activejob/lib/active_job/exceptions.rb

```diff
@@ -96,6 +99,10 @@ def discard_on(*exceptions)
           end
         end
       end
+
+      def abort_perform_on_retry(abort_perform = false)
```

Comment:
Should we make this macro not accepting values? 

If the default value from the `class_attribute` is false, we could make it so if someone calls `abort_perform_on_retry` at the job class level we simply set `abort_perform_on_retry_job` to true.

I think it simplifies using the macro as well.

```ruby
class HaltOnRetryJob < ActiveJob::Base
  abort_perform_on_retry

  def perform
    JobBuffer.add("Executing before retry_job was called")
    if executions < 2
      retry_job
      JobBuffer.add("Executing after retry_job was called")
    end
    JobBuffer.add("Job complete")
  end
end
```

Reply:
There _may_ be situations where currently existing jobs are relying on this ability. Explicitly passing the boolean would allow `false` to also be passed in the future if having this enabled ever becomes the default and certain jobs needed to opt-out for legacy reasons.